x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java

```diff
@@ -132,6 +159,37 @@ protected boolean skipExpression(Expression e) {
         }
     }
 
+    /**
+     * Local aggregation can only produce intermediate state that get wired into the global agg.
+     */
+    private static class LocalPropagateEmptyRelation extends PropagateEmptyRelation {
```

Comment:
The gist of this PR - override the global rule with a specific local rule that takes into account the intermediate aggregation and also takes care of handling the boolean types (seen).
This removes the weird logic downstream in the LocalExecutionPlanner as its much more contained.